### ISSUE TITLE
method设置为PROPFIND时，data参数被清空

### ISSUE THREAD

**Issue by thinking001** [flutterchina/dio#636]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps
请求method设置为PROPFIND时，data参数被清空，请问有什么方法可以解决
```dart
final response = await _client.request(
        url,
        data: data ?? Uint8List(0),
        options: Options(
          headers: headers ?? {},
          method: 'PROPFIND',
        ),
      );
```

源码中这个地方被限制了
```dart
//dio.dart文件
Future<Stream<Uint8List>> _transformData(RequestOptions options) async {
    var data = options.data;
    List<int> bytes;
    Stream<List<int>> stream;
    if (data != null &&
        ['POST', 'PUT', 'PATCH', 'DELETE'].contains(options.method)) {
```
